### DISCUSSION TITLE
Unable to get macOS application notarized

### DISCUSSION THREAD

**Discussion by rxliuli** [tauri-apps/tauri#8693] [Q&A]
I built a macOS application using Tauri and tried to publish it to the App Store, but it's been stuck for 2 hours... I received 3 errors and I'm not sure how to correct them.

```sh
Asset validation failed (90296)
App sandbox not enabled. The following executables must include the "com.apple.security.app-sandbox" entitlement with a Boolean value of true in the entitlements property list: [( "com.rxliuli.batch-rename.pkg/Payload/Batch Rename.app/Contents/MacOS/Batch Rename" )] Refer to App Sandbox page at https://developer.apple.com/documentation/security/app_sandbox for more information on sandboxing your app. (ID: f5bdedf9-cfa8-42e6-a26b-3f6520a3c02c)
```

I thought this error only required me to declare `com.apple.security.app-sandbox` in the Info.plist, but that didn't help.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
    <dict>
        <key>com.apple.security.app-sandbox</key>
        <true/>
    </dict>
</plist>
```

GitHub Link: https://github.com/rxliuli/batch-rename/blob/f91e95907d98ab8cda7271033fe403f8f6a60b9c/src-tauri/MyInfo.plist#L5-L6

```
Asset validation failed (90242)
The product archive is invalid. The Info.plist must contain a LSApplicationCategoryType key, whose value is the UTI for a valid category. For more details, see "Submitting your Mac apps to the App Store". (ID: febdd2d7-a513-4a80-8338-3229ad794e68)
```

Similarly, I have already configured `tauri.bundle.category: public.app-category.utilities`.

```
Asset validation failed (90869)
Invalid bundle. The “Batch Rename.app” bundle supports arm64 but not Intel-based Mac computers. Your build must include the x86_64 architecture to support Intel-based Mac computers. To support arm64 only, your macOS deployment target must be 12.0 or higher. For details, view: https://developer.apple.com/documentation/xcode/building_a_universal_macos_binary. (ID: 657554bd-aeee-43e8-a97f-638fae7fa528)
```

The build command I used is as follows:

```sh
pnpm tauri build --target universal-apple-darwin
productbuild --component './src-tauri/target/release/bundle/macos/Batch Rename.app' /Applications --sign "3rd Party Mac Developer Installer: NAME (ID)" ./release/batch-rename.pkg
```

GitHub Repo: https://github.com/rxliuli/batch-rename

**Comment by baimeidaxia** [comment]
Have you solved the problem? I have the same problem

**Comment by FabianLars** [comment]
> App sandbox not enabled

Create a new entitlements file like https://github.com/tauri-apps/tauri/blob/1.x/examples/updater/entitlements.plist (but with the sandbox entitlement) and include it in tauri.conf.json like this https://github.com/tauri-apps/tauri/blob/1.x/examples/updater/src-tauri/tauri.conf.json#L36 - Generally you do not add entitlements to the Info.plist file.

> LSApplicationCategoryType - I have already configured tauri.bundle.category: public.app-category.utilities

Can you try one of those values instead? iirc it was supposed to support your style too but maybe it doesn't https://tauri.app/v1/api/config/#bundleconfig.category 

> Invalid bundle. The “Batch Rename.app” bundle supports arm64 but not Intel-based Mac computers.

I assume this error is from _before_ you run the universal command? Because that's what i would have recommended here...

**Reply by u9646** [reply]
hi, i face App sandbox not enabled issue, and i tryed your demo code. but i can't solve this issue. this is what i do

1:  i created entitlements.plist file, and add to <root> dir. 
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>com.apple.security.app-sandbox</key>
    <true/>
    <key>com.apple.security.files.user-selected.read-write</key>
    <true/>
    <key>com.apple.security.print</key>
    <true/>
    <key>com.apple.security.network.client</key>
    <true/>
  </dict>
</plist>
```

then the project tree like this:
<root>
--entitlements.plist
--src-tauri/
----tauri.config.json

2: update tauri.config.json
```
"bundle": {
    "active": true,
    "targets": "all",
    "category": "Productivity",
    "macOS": {
      "signingIdentity": null,
      "entitlements": "../entitlements.plist",
      "frameworks": [],
      "minimumSystemVersion": "12.0",
      "exceptionDomain": ""
    },
    "icon": [
      "icons/icon.icns"
    ]
  },
```

3: run `pnpm run tauri build`

4: and i build success, src-tauri/target/release/bundle/macos/xxx.app generate successed。 
but xxx.app/Contents/ dir only find Info.plist, not found entitlements.plist file, i'am not sure it is expected?

5: then i try to sign, use 
```
codesign --sign "Apple Distribution: xx (xx)" "src-tauri/target/release/bundle/macos/xx.app/Contents/MacOS/xx-client"
```

this script done success

6: then to build .pkg file, i use
```
productbuild \
	--sign "3rd Party Mac Developer Installer: xx (xx)" \
	--component "src-tauri/target/release/bundle/macos/xx.app" "/Applications" \
	"xx.pkg"
```

and it doen success, and i get xx.pkg

7: then i use Transporter upload this xx.pkg, upload successed, and then run Verify, then i still get this error
![image](https://github.com/user-attachments/assets/6a88d55f-f919-46cf-b2a5-f3c80c9d6e97)


could you can help me to solve this issue, thank you very much

**Reply by DJBurb** [reply]
Did you ever resolve this issue?  I am having the exact same thing happen to me.

**Reply by ascoolarobban** [reply]
Same here, tried most on the internet and no luck, so please if anybody solved this id love to know how

**Comment by 771991673** [comment]
same here, I stuck with it for three days TT

**Reply by richtermb** [reply]
how did it end up?

**Comment by alexfromapex** [comment]
You guys need to make an `entitlements.plist` file, not `info.plist` , then in your tauri.conf.json:

Add this:

```
... rest of your file ...
    "macOS": {
      "signingIdentity": "-",
      "entitlements": "entitlements.plist",
      ... rest of your file ...
```
